pkg/controller/deployment/deployment_controller.go

```diff
@@ -491,7 +491,11 @@ func (dc *DeploymentController) syncRollingUpdateDeployment(deployment extension
 	// Sync deployment status
 	totalReplicas := deploymentutil.GetReplicaCountForRCs(allRCs)
 	updatedReplicas := deploymentutil.GetReplicaCountForRCs([]*api.ReplicationController{newRC})
-	if deployment.Status.Replicas != totalReplicas || deployment.Status.UpdatedReplicas != updatedReplicas {
+	availablePods, err := deploymentutil.GetAvailablePodsForRCs(dc.client, allRCs, deployment.Spec.Strategy.RollingUpdate.MinReadySeconds)
+	if err != nil {
+		return fmt.Errorf("failed to count ready pods: %v", err)
+	}
+	if deployment.Status.Replicas != totalReplicas || deployment.Status.UpdatedReplicas != updatedReplicas || deployment.Status.AvailableReplicas != availablePods {
```

Comment:
This is extracted in #19733 as well so I kept the same name, `syncDeploymentStatus`.